45 main header bouwen voor desktop

Jason2426 opened the pull request:
# Main header

Built  #45, I also made it responsive without having to write styling in media queries. I also removed some of the categories in the previous header and added a category icon instead which will bring the user to the page with all the categories because I thought the old design was less efficient. 

**To check :** 
- HTML if there are improvements for accessibility and if its semantic 
- CSS improvements for readability or chances to remove unnecessary lines 

The functionality of the theme button still has to be made #39 and can now be made.

<img width="1340" alt="Screenshot 2024-05-26 at 01 49 26" src="https://github.com/fdnd-task/pleasurable-ui/assets/143999883/7ff5edb8-1780-4860-a5c3-379c3a6e1c46">

christoph3r3w commented on public/styles/header.css:
```diff
@@ -0,0 +1,203 @@
+    width: 220px;
+}
+
+@supports (display: flex){
```
@supports is mostly used when a feature is new or is not widely supported by every browser. but most of the properties you used in this section are commonly used so @supports is not needed

christoph3r3w commented:
It's nice when a component can adapt to any screen size, but it's still good to use media queries or container-queries for multiple breaking points to prevent things like this: 

![image](https://github.com/christoph3r3w/pleasurable-ui/assets/144007933/2fcfefb9-de7e-410c-a855-1f62f10cc780)
> top left corner not aligning 

![image](https://github.com/christoph3r3w/pleasurable-ui/assets/144007933/b201fe92-0389-4ef2-a389-35fc45ed4575)
![image](https://github.com/christoph3r3w/pleasurable-ui/assets/144007933/dbc22623-8b0f-43da-9392-a1a43a929865)
> there is no dynamic gap between elements

christoph3r3w commented on views/partials/header.ejs:
```diff
@@ -48,4 +48,71 @@
+				Live date here..
+			</div>
+			<div class="second-col">
+				<picture>
```
It is always good to have an element in the header that refreshes the page or leads the user back to the home page.  since users are used to having this feature with the logo i would recommend to do the same.

christoph3r3w replied:
you can do this by wrapping the logo in a link. and the link in a div

christoph3r3w reviewed (changes requested):
- [x] logo
- [ ] media or container queries
- [x] @supports

Jason2426 commented:
# Fixes 

- Fixed logo route for detailpage 
- Removed unecassary @supports rule
- Added styling in a @media query so it looks nice 

? detailpage has a huge random logo on it and spacing is weird but I didnt change anything in those css files ? 

![Screenshot 2024-05-26 at 15 25 18](https://github.com/christoph3r3w/pleasurable-ui/assets/143999883/94393084-19e6-4ebd-b171-439deeaf3e52)

christoph3r3w commented:
> # Fixes
> * Fixed logo route for detailpage
> * Removed unecassary @supports rule
> * Added styling in a @media query so it looks nice
> 
> ? detailpage has a huge random logo on it and spacing is weird but I didnt change anything in those css files ?
> 
> ![Screenshot 2024-05-26 at 15 25 18](https://private-user-images.githubusercontent.com/143999883/333882543-94393084-19e6-4ebd-b171-439deeaf3e52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3NDEwOTksIm5iZiI6MTcxNjc0MDc5OSwicGF0aCI6Ii8xNDM5OTk4ODMvMzMzODgyNTQzLTk0MzkzMDg0LTE5ZTYtNGViZC1iMTcxLTQzOWRlZWFmM2U1Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNlQxNjI2MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2NjMzFjNzIxYjYzYjMzMGUzNjE4OWM5MWY2Y2QwMGY2NGRiMmE5MmFjMjFiNTU3MDdhYmJmMTQ4MWJiYWM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.AuAzt8AHARPgNjPHqHTu2mrsdFmWMzROs1GDm8xcR-Y)

yeah, i had trouble with it too. it's because Redpers added a new layout for that article. bu i think i fixed it in #47
